src/hatch/cli/python/install.py

```diff
@@ -41,10 +43,7 @@ def install(app: Application, *, names: tuple[str, ...], private: bool, update:
     from hatch.python.distributions import ORDERED_DISTRIBUTIONS
     from hatch.python.resolve import get_distribution
 
-    shells = []
-    if not private and not app.platform.windows:
```

Comment:
I'm going to revert this part because we want this conditional to prevent the shell detection logic for performance reasons.